Implement feature to print the logs of failed rules

sanjaynagi opened the pull request:
Hey Vince! 

Nice tool! I wanted the ability to automatically print to stdout the failed rule logs from a snakemake log file, so I added it as a feature (again with the help of Claude AI). 

By specifying `snklog failed` or `snklog f`, snklog will print all of the rule logs that came from failed rules. By default, it will select the most recent snakemake log. You can also use a specific log with the identifier (M1 , S1 etc), ie `snklog f M1`. 

Have not extensively tested it but it seems to be working well!

sanjaynagi committed:
adds feature to print failed rule logs

sanjaynagi committed:
add to readme

sanjaynagi commented:
oh man this is actually so lovely to use, I should have done this years ago!!

vsbuffalo commented:
Thanks so much Sanjay! I would be happy to incorporate this. However, I was testing it out on some recent jobs I had (which I am pretty sure had failures) and I didn't see any crop up. Do you have an example log that this works on you could share?

sanjaynagi commented:
thats awesome. 

Hmmmm, don't know whether its this, but I've just noticed that it will report there are no failed rules if a log file isn't specified per rule. I'll sort that so it at least prints the failed rules. 

Here are the logs that should work, you'll need both logs in this structure. 

logs/[PerGeneFstPBSDxyPi.log](https://github.com/user-attachments/files/16433220/PerGeneFstPBSDxyPi.log)
.snakemake/log/[2024-07-29T105026.063919.snakemake.log](https://github.com/user-attachments/files/16433223/2024-07-29T105026.063919.snakemake.log)

Heres what it looks like for me :)

![image](https://github.com/user-attachments/assets/23b9fd9c-97ca-40b3-a31d-e897dfccc413)

I'll do some more testing though too.

sanjaynagi committed:
adds fv verbose option which prints logs of snakemake jobs

sanjaynagi commented:
Should work better now - I've fixed the behaviour which missed rules without logs and I've changed the behaviour so that the `snklog f` option just shows which snakemake jobs have failed, and added a new option `snklog fv` (v for verbose) which shows which jobs have failed AND prints the specific snakemake job log if its present :)
